packages/wallet/backend/src/outgoingPayment/service.ts

```diff
@@ -1,124 +1,44 @@
+import { RafikiClient } from '@/rafiki/rafiki-client'
+import { Transaction } from '@/transaction/model'
 
 interface IOutgoingPaymentService {
-  create: (
-    userId: string,
-    paymentPointerId: string,
-    amount: number,
-    isReceive: boolean,
-    receiver: string,
-    description?: string
-  ) => Promise<Transaction>
+  create: (quoteId: string) => Promise<Transaction>
 }
 
 interface OutgoingServiceDependencies {
-  accountService: AccountService
   rafikiClient: RafikiClient
   incomingPaymentService: IncomingPaymentService
 }
 
 export class OutgoingPaymentService implements IOutgoingPaymentService {
   constructor(private deps: OutgoingServiceDependencies) {}
 
-  async create(
-    userId: string,
-    paymentPointerId: string,
-    amount: number,
-    isReceive: boolean,
-    receiver: string,
-    description?: string
-  ): Promise<Transaction> {
-    const existingPaymentPointer = await PaymentPointer.query().findById(
-      paymentPointerId
-    )
-
-    if (!existingPaymentPointer || !existingPaymentPointer.active) {
-      throw new BadRequest('Invalid payment pointer')
-    }
-
-    const { assetId, assetCode } =
-      await this.deps.accountService.findAccountById(
-        existingPaymentPointer.accountId,
-        userId
-      )
-    const balance = await this.deps.accountService.getAccountBalance(
-      userId,
-      assetCode
-    )
-
-    if (Number(balance) < amount) {
-      throw new BadRequest('Not enough funds in account')
-    }
+  async create(quoteId: string): Promise<Transaction> {
+    const quote = await this.deps.rafikiClient.getQuote(quoteId)
 
-    const asset = await this.deps.rafikiClient.getAssetById(assetId)
-    if (!asset) {
-      throw new NotFound()
-    }
+    const paymentPointerId = quote.paymentPointerId
 
-    let paymentUrl = receiver
-    if (!incomingPaymentRegexp.test(receiver)) {
-      paymentUrl = await this.createReceiver(
-        isReceive ? BigInt(amount * 10 ** asset.scale) : null,
-        asset,
-        receiver,
-        description,
-        new Date(Date.now() + 1000 * 60).toISOString()
+    const incomingPayment =
+      await this.deps.incomingPaymentService.getPaymentDetailsByUrl(
+        quote.receiver
       )
-    }
+    const { description, assetCode, value } = incomingPayment
 
-    const quote = await this.deps.rafikiClient.createQuote({
-      paymentPointerId,
-      receiver: paymentUrl,
-      asset,
-      amount: isReceive ? undefined : BigInt(amount * 10 ** asset.scale)
-    })
     const payment = await this.deps.rafikiClient.createOutgoingPayment(
       paymentPointerId,
-      quote.id,
+      quoteId,
       description
     )
 
     return Transaction.query().insert({
-      paymentPointerId: existingPaymentPointer.id,
+      paymentPointerId: paymentPointerId,
       paymentId: payment.id,
-      assetCode: asset.code,
-      value: BigInt(amount * 10 ** asset.scale),
+      assetCode,
+      value: BigInt(value),
```

Comment:
should it be `BigInt(amount * 10 ** asset.scale)`

Reply:
Hmm i got it from the incoming payment. But I guess i should get it from the sendAmount from the quote.